src/containers/CommunityModal/CommunityModal.tsx

```diff
@@ -0,0 +1,170 @@
+import * as Yup from 'yup';
+import React, { useState, useRef } from 'react';
+import { useFormik } from 'formik';
+import CommunityModalImageUpload from '@src/containers/CommunityModal/CommunityModalImageUpload/CommunityModalImageUpload';
+import {
+  PreviewImage,
+  CardWrapper,
+  ModalErrorMessage,
+} from '@src/containers/CommunityModal/CommunityModal.style';
+import {
+  editCommunityPostApi,
+  postCommunityPostApi,
+} from '@src/apis/community';
+import IconButton from '@mui/material/IconButton';
+import {
+  Button,
+  Modal,
+  Avatar,
+  CardHeader,
+  CardContent,
+  TextField,
+} from '@mui/material';
+import ClearIcon from '@mui/icons-material/Clear';
+import { CommunityModalType } from '@interfaces/community';
+
+const SUPPORTED_FORMATS = ['image/jpg', 'image/jpeg', 'image/gif', 'image/png'];
+
+function CommunityModal({
+  postId,
+  nickname,
+  profileImageUrl,
+  modalType,
+  isOpen,
+  onClose,
+}: CommunityModalType) {
+  const [previewUrl, setPreviewUrl] = useState('');
+  const [isImageSize, setImageSize] = useState(false);
+  const exitRef = useRef<any>(null);
+
+  const handleImageUpload = (e: React.ChangeEvent<any>) => {
+    if (!e.currentTarget.files[0]) return;
+    if (e.currentTarget.files[0].size > 1024 * 1024) {
+      setImageSize(true);
+      return;
+    }
+    const objectUrl = URL.createObjectURL(e.currentTarget.files[0]);
+    previewUrl && URL.revokeObjectURL(previewUrl);
+    setImageSize(false);
+    setPreviewUrl(objectUrl);
+
+    formik.setFieldValue('postImage', e.currentTarget.files[0]);
+  };
+
+  const handlePost = async (values: {
+    contents: string;
+    postImage: string;
+  }) => {
+    const postFormData = new FormData();
+
+    postFormData.append('contents', values.contents);
+    postFormData.append('postImage', values.postImage);
+
+    if (modalType === 'CREATE') {
+      await postCommunityPostApi({
+        url: 'posts',
+        postData: postFormData,
+      });
+    } else if (modalType === 'EDIT') {
+      await editCommunityPostApi({
+        url: `posts/${postId}`,
+        postData: postFormData,
+      });
+    }
+  };
+
+  //formik
+  const formik = useFormik({
+    initialValues: {
+      contents: '',
+      postImage: '',
+    },
+    onSubmit: (values, { setSubmitting }) => {
+      setSubmitting(true);
+      values.contents = '';
+      setPreviewUrl('');
+      handlePost(values);
+      exitRef.current.click();
+      setSubmitting(false);
+    },
+    validationSchema: Yup.object({
+      contents: Yup.string()
+        .max(500, '1,000자를 넘을 수 없습니다.')
```

Comment:
```suggestion
        .max(1000, '1,000자를 넘을 수 없습니다.')
```
max length가 잘못 지정된 것 같습니다.

Reply:
오탈자가 있었네요 ㅎㅎㅎㅎㅎ